micrometer-java21/src/main/java/io/micrometer/java21/instrument/binder/jdk/VirtualThreadMetrics.java

```diff
@@ -90,12 +124,16 @@ public void close() {
         recordingStream.close();
     }
 
-    private record RecordingConfig(Duration maxAge, long maxSizeBytes, Duration pinnedThreshold) {
+    public void setActiveMetricEnabled(boolean activeMetricEnabled) {
+        this.activeMetricEnabled = activeMetricEnabled;
```

Comment:
Having a setter creates an implicit invariant that this needs to be called before `bindTo` or it will not have any effect. That may surprise some users, and I also don't want to establish it has a precedent where we'd similarly have to add a setter for any future additions. What about making this an option on `RecordingConfig` instead? It was somewhat intentional it wasn't public so that if things like this came up we'd be able to more easily change it. Once we make it public, we need to take care to not break backward compatibility.